doc/06-distributed-monitoring.md

```diff
@@ -3230,6 +3230,31 @@ information/pki: Writing certificate to file 'icinga2-satellite1.localdomain.crt
 
 Copy and move these certificates to the respective instances e.g. with SSH/SCP.
 
+#### External CA/PKI
+
+Neither the above commands, nor their automatic counterparts in the Icinga
+cluster do anything special during certificate issuance. I.e. Icinga
+isn't the only possible source of the certificates it uses. E.g.
+`openssl req/x509 ...` may be used as well as long as the leaf certificates' CN
+and SAN match the endpoint names. Pretty much everything else is limited only by
+your imagination and the oldest OpenSSL version of two Icinga nodes connected to
+each other. E.g. the following works:
```

Comment:
> One shouldn't use external CAs if doesn't know what to do though.

Actually a reason not to make this more user friendly. 😅

Reply:
I was referring to the examples below and not this opening text. This is still a technical documentation, so you have to comply with certain standards (avoiding the use of *i.e* or *e.g*).

Reply:
@julianbrost What's your opinion?

Reply:
In general, it's a good idea to write down what Icinga 2 expects from certificates and what it issues by itself. However, I think I'd rather place this somewhere into the technical details documentation instead of here and telling users so be "limited only by [their] imagination". It should be more of a "only do this if you know what you're doing and don't expect much help" thing not an invitation to generate special certificates and then be surprised that they weren't (properly) renewed by Icinga 2.

Reply:
I'd even agree with the second half of your comment, but isn't _Manual Certificate Creation_ exactly where users would expect such? Users already can't find something in our docs w/o a search engine!

Reply:
> Actually a reason not to make this more user friendly. 😅

> exactly where users would expect such

Which of these two now? Currently it's a "be limited only by your imagination and good luck with that" for end users. I'd be happy with not directly presenting it to most standard users but moving it to the more technical documentation that's more for developers and expert users.

If this is supposed to become standard user documentation, this at the very least needs a warning that custom certificates may not be renewed by Icinga 2 and that one has to take care of this externally. By the way, did you test what happens when Icinga 2 would attempt to renew an externally issued certificate that does not closely resemble the ones issued by Icinga 2?

Reply:
As I was asked to put my two euro cents in, I am generally in favor of this PR, as it both (loosely) describes Icinga 2's own requirements for certificates as well as how to realize a "SSL added and removed here! :-)"-setup required by some corporations.

However, I would advise adding more details and moving it to "12-icinga2-api" or another later section, to not confuse or overload first time users. Furthermore, a hint how to monitor the certificate expiration - link to ITL, check_ssl_cert, for example - would be good, as otherwise people tend to forget those things, even if noted that Icinga 2 itself does not do a thing.

Reply:
> However, I would advise adding more details and moving it to "12-icinga2-api" or another later section, to not confuse or overload first time users.

I'm afraid this naturally belongs to _Manual Certificate Creation_. That section can be seen as a bit advanced by itself and its parent section is even called _Advanced Hints_, so... 🤷‍♂️ (Have I already mentioned there's a big warning now in here?)